scripts/simplify.py

```diff
@@ -0,0 +1,215 @@
+			# Check if field has a value
+			if not trace.getAttribute(allowed_attr):
+				continue
+		
+			new_trace.setAttribute(allowed_attr, trace.getAttribute(allowed_attr))
+
+		# Assign default trace style
+		new_trace.setAttribute('style', default_trace_style)
+
+		# Add UUID if needed
+		if not new_trace.getAttribute('pcbre:uuid'):
+			new_trace.setAttribute('pcbre:uuid', str(uuid.uuid4()))
+
+		# Sort attributes
+		sort_element_attr(new_trace)
+
+		new_layer.appendChild(new_trace)
+
+	# Sort elements
+	new_layer.childNodes.sort(key=lambda x: x.getAttribute('pcbre:uuid'))
+
+def main(argv):
+    # Load and parse SVG
+	input_xml = minidom.parse(argv[0])
+	input_svg = input_xml.getElementsByTagName('svg')[0]
+
+	# Get 'layers' from SVG
+	layers = []
+
+	for element in input_xml.getElementsByTagName('g'):
+		# Sanity check
+		if element.getAttribute('inkscape:groupmode') != 'layer':
+			raise Exception('None layer groups are not supported at this time!')
+
+		if not element.getAttribute('pcbre:uuid'):
+			print('WARNING: No UUID assigned to layer. Skipping layer', element.getAttribute('inkscape:label'))
+			continue
+
+		layers += [element]
+
+	# Sort layers
+	layers.sort(key=lambda x: x.getAttribute('pcbre:uuid'))
+
+	# Create our 'clean' SVG object
+	clean_xml = minidom.Document()
+	clean_svg = clean_xml.createElement('svg')
+
+	# Recreate each allowed attribute
+	svg_allowed_attrs = ['width', 'height', 'viewBox']
+	for svg_allowed_attr in svg_allowed_attrs:
+		# Check if field has a value
+		if not input_svg.getAttribute(svg_allowed_attr):
+			continue
+	
+		clean_svg.setAttribute(svg_allowed_attr, input_svg.getAttribute(svg_allowed_attr))
+
+	# Add back XML namespcaes
+	input_svg_attributes = list(input_svg.attributes.keys()) if input_svg.attributes else []
+	for attribute in input_svg_attributes:
+		if attribute.startswith('xmlns:'):
+			clean_svg.setAttribute(attribute, input_svg.getAttribute(attribute))
+
+	# Sort attributes
+	sort_element_attr(clean_svg)
+
+	# Process each layer
+	layer_allowed_attrs = ['inkscape:label', 'transform', 'pcbre:uuid']
+	for layer in layers:
+		print('Processsing layer:', layer.getAttribute('inkscape:label'))
+
+		# Create group element
+		new_layer = clean_xml.createElement('g')
+
+		# Set default attributes
+		for layer_attr, value in default_layer_attrs.items():
+			new_layer.setAttribute(layer_attr, value)
+
+		# Recreate each allowed attribute in the newly created element
+		for svg_allowed_attr in layer_allowed_attrs:
+			# Check if field has a value
+			if not layer.getAttribute(svg_allowed_attr):
+				continue
+		
+			new_layer.setAttribute(svg_allowed_attr, layer.getAttribute(svg_allowed_attr))
+
+		# Add UUID if needed
+		if not element.getAttribute('pcbre:uuid'):
+			element.setAttribute('pcbre:uuid', str(uuid.uuid4()))
+
+		# Sort attributes
+		sort_element_attr(new_layer)
+
+		# Sub elements of current layer
+		if layer.getAttribute('inkscape:label') in ['Board Bottom', 'Board Top']:
```

Comment:
We should not depend on inkscape features as some people might want to use other tools (we don't know how large this project will eventually get, especially if we might do other revisions of the motherboard or other PCBs in general).
Making the tool generic will also help with making a generic workflow that other communities can re-use.

Reply:
> We should not depend on inkscape features as some people might want to use other tools (we don't know how large this project will eventually get, especially if we might do other revisions of the motherboard or other PCBs in general).
> Making the tool generic will also help with making a generic workflow that other communities can re-use.

Fixed in f3277365787f9aa8112f77eaeca83fb2549c81c3

This commit removes all of the namespaces expect for xlink which is used for linking an external image to an element. 

Instead of using Inkscape labels we are now using the element ID attribute. This appears to work correctly in Inkscape as the label for the object.

Also, removed 'inkscape:groupmode'. The side-effect of this that there are no longer 'Layers' in Inkscape, but the same functionality exists in 'Objects'.

The attribute 'inkscape:connector-curvature' was removed. I did not spend a lot of time investing this, but it appears to have no ill-effect on the output.

> A solution would be to have 1 file per layer, or check if SVG has some layer support itself.
> We could also have some preprocessor / postprocessor which splits or combines the different files into one file for inkscape. However, I don't think that's necessary anyway, as people can work top and bottom independently.

I feel like this outside of the scope this pull request and should be addressed in #8. I will keep this into consideration though.